server/app/graphql/types/gene_claim_category_type.rb

```diff
@@ -4,19 +4,37 @@ class GeneClaimCategoryType < Types::BaseObject
     field :name, String, null: false
 
     field :gene_claims, [Types::GeneClaimType], null: false
-    field :genes, Types::GeneType.connection_type, null: false
-    field :gene_count, Int, null:false
 
     def gene_claims
       Loaders::AssociationLoader.for(GeneClaimCategory, :gene_claims).load(object)
     end
 
-    def genes
-      Loaders::AssociationLoader.for(GeneClaimCategory, :genes).load(object)
+    class GeneCategoryResult < Types::BaseObject
+      field :name, String, null: false
+      field :concept_id, String, null: false
+      field :long_name, String, null: false
+      field :source_db_names, [String],  null: false
     end
 
-    def gene_count
-      object.genes.count
+    field :genes, GeneCategoryResult.connection_type, null: false do
+      argument :source_names, [String], required: false
+      argument :category_name, String, required: true
+    end
+
+    def genes(source_names: [], category_name: '')
```

Comment:
@acoffman dumb question -- how do we reuse the `name` param passed to the type instead of providing it twice? nbd but it ends up making the query look a little silly:

```
    query geneClaimCategory($categoryName: String!, $sourceDbNames: [String!]!) {
      geneClaimCategory(name: $categoryName) {
        name
        genes(categoryName: $categoryName, sourceNames: $sourceDbNames) {
          edges {
            node {
              name
              conceptId
              longName
              sourceDbNames
            }
          }
        }
      }
    }
```

Reply:
Are you asking how to avoid supplying `$categoryName` to both the `geneClaimCategory` and `genes` portions of the query?

That would be done in the back end resolver, first by having the `genes` resolver check if it has a parent or not and setting the default `scope` appropriately and then, depending on the structure of the query pulling the argument from its parent context. If that's unclear I'm happy to take a look at the code/make a PR